draft-ietf-intarea-proxy-config.md

```diff
@@ -154,18 +154,22 @@ Each proxy is defined by a proxy protocol, a proxy location (i.e., a hostname an
 
 This document defines two mandatory keys for the sub-dictionaries in the
 `proxies` array, `protocol` and `proxy`. There are also optional keys, including
-`alpn`, and destination accessibility keys defined in {{destinations}}.
-Other optional keys can be added to the dictionary
-to further define or restrict the use of a proxy. Clients that do not
-recognize or understand a key in a proxy sub-dictionary MUST ignore the entire
-proxy definition, since the proxy might be only applicable for particular
-uses. These keys are registered in an IANA registry, defined in {{proxy-info-iana}}.
+`alpn`, `mandatory`, and destination accessibility keys defined in {{destinations}}.
+Other optional keys can be added to the dictionary to further define or restrict the
+use of a proxy.
 
 | JSON Key | Optional | Description | Type | Example |
 | --- | --- | --- | --- | --- |
 | protocol | No | The protocol used to communicate with the proxy | String | "connect-udp" |
 | proxy | No | String containing the URI template or hostname and port of the proxy, depending on the format defined by the protocol | String | "https://proxy.example.org:4443/masque{?target_host,target_port}" |
 | alpn | Yes | An array of Application-Layer Protocol Negotiation protocol identifiers | Array of Strings | ["h3","h2"] |
+| mandatory | Yes | An array of optional keys that client must understand and process to use this proxy | Array of Strings | ["matchDomains"] |
+
+The value of the `mandatory` key is a list of keys that the client must understand and process to be
+able to use the proxy. A client that does not understand a key from the list or cannot fully process
+the value of a key from the list MUST ignore the entire proxy definition. The list can contain
+only keys that are registered in an IANA registry, defined in {{proxy-info-iana}} and that are marked
+as optional.  The `mandatory` list MUST NOT include any entries that are not present in the sub-dictionary.
```

Comment:
Sorry, this is on me. I was not precise enough.

Can you move this below alpn, so that the keys are discussed in the same order as in the table.
Thanks!